elements/pl-units-input/pl-units-input.py

```diff
@@ -0,0 +1,468 @@
+            raise ValueError('Cannot set parameter "rtol" in unit agnostic grading.')
+
+        partial_credit = pl.get_float_attrib(
+            element, "magnitude-partial-credit", MAGNITUDE_PARTIAL_CREDIT_DEFAULT
+        )
+
+        if partial_credit is not None and not (0.0 <= partial_credit <= 1.0):
+            raise ValueError(
+                f'"magnitude-partial-credit" must be in the range [0.0, 1.0], not {partial_credit}'
+            )
+
+        correct_answer_parsed = ureg.Quantity(correct_answer)
+
+        if (correct_answer_parsed is not None) and (
+            not correct_answer_parsed.check(parsed_atol.dimensionality)
+        ):
+            raise ValueError(
+                f"Correct answer has dimensionality: {correct_answer_parsed.dimensionality}, "
+                f"which does not match atol dimensionality: {parsed_atol.dimensionality}."
+            )
+    else:
+        if not parsed_atol.dimensionless:
+            raise ValueError(
+                f'atol parameter "{atol}" may only have units in units agnostic grading.'
+            )
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
+    element = lxml.html.fragment_fromstring(element_html)
+    name = pl.get_string_attrib(element, "answers-name")
+    label = pl.get_string_attrib(element, "label", LABEL_DEFAULT)
+    suffix = pl.get_string_attrib(element, "suffix", SUFFIX_DEFAULT)
+    display = pl.get_enum_attrib(element, "display", DisplayType, DISPLAY_DEFAULT)
+    size = pl.get_integer_attrib(element, "size", SIZE_DEFAULT)
+    comparison = pl.get_enum_attrib(
+        element, "comparison", uu.ComparisonType, COMPARISON_DEFAULT
+    )
+    grading_mode = pl.get_enum_attrib(
+        element, "grading-mode", GradingMode, GRADING_MODE_DEFAULT
+    )
+    show_info = pl.get_boolean_attrib(element, "show-help-text", SHOW_HELP_TEXT_DEFAULT)
+
+    raw_submitted_answer = data["raw_submitted_answers"].get(name, None)
+    partial_scores = data["partial_scores"].get(name, {})
+    score = partial_scores.get("score")
+
+    if data["panel"] == "question":
+        editable = data["editable"]
+
+        # Get info strings
+        with open(UNITS_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
+            info = chevron.render(
+                f,
+                {"format": True, "only_units": grading_mode is GradingMode.ONLY_UNITS},
+            ).strip()
+
+        if pl.has_attrib(element, "placeholder"):
+            placeholder_text = pl.get_string_attrib(element, "placeholder")
+        elif grading_mode is GradingMode.ONLY_UNITS:
+            placeholder_text = "Unit"
+        elif grading_mode is GradingMode.WITH_UNITS:
+            atol = pl.get_string_attrib(element, "atol", ATOL_DEFAULT)
+            placeholder_text = f"Number (atol={atol}) + Unit"
+
+        # Grading mode must be units fixed at this point,
+        # now can case based on comparison
+        elif comparison is uu.ComparisonType.RELABS:
```

Comment:
I would prefer a nested if statement to case based on a different variable, with no comments, rather than having a complicated comment to explain the odd programming pattern. I think its more readable that way.

Reply:
Done!